org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/DocCommentParser.java

```diff
@@ -719,9 +719,87 @@ protected boolean parseReturn() {
 	}
 
 	@Override
-	protected boolean parseMarkdownLinks() {
-		// TODO Auto-generated method stub
-		return false;
+	protected boolean parseMarkdownLinks(int previousPosition) throws InvalidInputException {
+		boolean valid = false;
+		// The markdown links can come in single [] or pair of [] with no space between them
+		// We are here after we have seen [
+		// Look for closing ] and then an option [
+		// immediately without any other characters, including whitespace
+		// If there are two [], then the first one becomes the link text
+		// and the second one is the reference
+		// in case of just one [], then that is the reference
+		int start = this.index;
+		char currentChar = readChar();
+		int tStart = previousPosition;
+		int tEnd = -1;
+		loop: while (this.index < this.scanner.eofPosition) {
+			switch(currentChar) {
+				case '\\':
+					char c = peekChar();
+					if (c == '[' || c == ']') {
+						readChar();
+					}
+					break;
+				case ']':
+					if (peekChar() == '[') {
+						tStart = start;
+						tEnd = this.index - 1;
+						currentChar = readChar();
+						start = this.index;
+					} else {
+						int eofBkup = this.scanner.eofPosition;
+						this.scanner.eofPosition = this.index - 1;
+						this.scanner.resetTo(start, this.javadocEnd);
+						this.inlineTagStarted = true;
+						this.inlineTagStart = previousPosition;
+						this.tagValue = TAG_LINK_VALUE;
+						int indexBkup = this.index;
+						valid = parseReference(true);
+						this.index = indexBkup;
+						// This creates a two level structure. The @link tag is added to
+						// another tag element, which gets added to the astStack
+						// Both tag elements must get the same source range.
+						TagElement previousTag = (TagElement) this.astStack[this.astPtr];
+						int parentStart = previousTag.getStartPosition();
+						int length = this.index - previousPosition + 1;
+						previousTag.setSourceRange(parentStart, this.index - parentStart + 1);
+						List fragments = previousTag.fragments();
+						int size = fragments.size();
+						if (size == 0) {
+							// no existing fragment => just add the element
+							TagElement inlineTag = this.ast.newTagElement();
+							fragments.add(inlineTag);
+							previousTag = inlineTag;
+						} else {
+							// If last fragment is a tag, then use it as previous tag
+							ASTNode lastFragment = (ASTNode) fragments.get(size-1);
+							if (lastFragment.getNodeType() == ASTNode.TAG_ELEMENT) {
+								//lastFragment.setSourceRange(lastFragment.getStartPosition(), length);
```

Comment:
@jarthana  I have a test case in JDT/UI, which succeeds if I enable this line, but with a length one less, i.e., `length = this.index - previousPosition;`
I can't give precise math of why this should be better, just that without that change after a link the closing ']' appears as part of subsequent text.